src/app/content/components/TableOfContents/keyboardSupport.hook.tsx

```diff
@@ -0,0 +1,183 @@
+import { HTMLElement, Element } from '@openstax/types/lib.dom';
+import { LinkedArchiveTreeNode } from '../../types';
+import React from 'react';
+
+export interface KeyboardSupportProps {
+  event: React.KeyboardEvent<HTMLAnchorElement>;
+  item: LinkedArchiveTreeNode;
+  isOpen?: boolean;
+  onSelect: () => void;
+}
+
+export const useKeyboardSupport = () => {
+
+  const getTreeItems = () => {
+    const treeItems = Array.from(document?.querySelectorAll('a[role="treeitem"]') as ArrayLike<Element>);
+    const treeItemResults = treeItems.filter((el) => {
+      const style = window?.getComputedStyle(el);
+      return (style?.display !== 'none');
+    });
+
+    return treeItemResults;
+  };
+
+  const focusNextTreeItem = (filteredTreeItems: Element[], currentItemIndex: number) => {
+    (document?.querySelector(`[id="${filteredTreeItems[currentItemIndex + 1]?.id}"]`) as HTMLElement)?.focus();
+  };
+
+  const focusPrevTreeItem = (filteredTreeItems: Element[], currentItemIndex: number) => {
+    (document?.querySelector(`[id="${filteredTreeItems[currentItemIndex - 1]?.id}"]`) as HTMLElement)?.focus();
+  };
+
+  const focusParent = (filteredTreeItems: Element[], currentItemIndex: number) => {
+    (document?.querySelector(
+      `[id="${filteredTreeItems[currentItemIndex]?.id}"]`)
+      ?.parentElement?.parentElement
+      ?.previousElementSibling as HTMLElement)
+      ?.focus();
+  };
+
+  const focusFirstTreeItem = (filteredTreeItems: Element[]) => {
+    (document?.querySelector(`[id="${filteredTreeItems[0]?.id}"]`) as HTMLElement)?.focus();
+  };
+
+  const focusLastTreeItem = (filteredTreeItems: Element[]) => {
+    (document?.querySelector(`[id="${filteredTreeItems[filteredTreeItems.length - 1]?.id}"]`) as HTMLElement)?.focus();
+  };
+
+  const focusOnTab = (filteredTreeItems: Element[], shiftKey: boolean) => {
+    const pageElements = Array.from(
+      document?.querySelectorAll('input, button, select, textarea, a[href]') as ArrayLike<Element>
+    );
+    let index: number;
+    if (shiftKey) {
+      focusFirstTreeItem(filteredTreeItems);
+      index = pageElements.indexOf(document?.activeElement as HTMLElement) - 1;
+    } else {
+      focusLastTreeItem(filteredTreeItems);
+      index = pageElements.indexOf(document?.activeElement as HTMLElement) + 1;
+    }
+    (pageElements[index] as HTMLElement)?.focus();
+  };
+
+  const searchTreeItem = (filteredTreeItems: Element[], currentItemIndex: number, key: string) => {
+    /*
+      According Keyboard Support for Navigation Tree, the search starts with treeItems
+      next to current treeItem and if there is no result, it will look for previous treeItems
+    */
+    const firstSearchOptions = filteredTreeItems.slice(currentItemIndex + 1, filteredTreeItems.length);
+    const secondSearchOptions = filteredTreeItems.slice(0, currentItemIndex);
+    let result;
+    for (const treeItem of secondSearchOptions) {
```

Comment:
I'm pretty sure you can concat the search options together and do a single `find`.
```
  const searchOptions = filteredTreeItems.slice(currentItemIndex + 1, filteredTreeItems.length)
    .concat(filteredTreeItems.slice(0, currentItemIndex));
  const result = searchOptions.find((treeItem) =>
    treeItem.textContent?.trim().toLowerCase().startsWith(key.toLowerCase())
  );
```